src/org/zaproxy/zap/extension/jxbrowser/ZapBrowserPanel.java

```diff
@@ -87,6 +89,15 @@ public Browser getBrowser() {
                 browser = new Browser();
             }
         }
+        if (LOGGER.isDebugEnabled()) {
+            if (browser.getConsoleListeners().isEmpty()) {
+                browser.addConsoleListener(new ConsoleListener() {
```

Comment:
This can be done in the if `browser == null` instead (that already ensures the listener is only added once).